app/commands/fetch_datasource_expenses.py

```diff
@@ -2,8 +2,10 @@
 import logging
 from collections.abc import Sequence
 from datetime import UTC, datetime
+from typing import Annotated
 
 import typer
+from dateutil.relativedelta import relativedelta  # type: ignore[import-untyped]
```

Comment:
```suggestion
from dateutil.relativedelta import relativedelta
```

Instead of ignoring, we can add the types for dateutil to the dev dependencies https://pypi.org/project/types-python-dateutil/